src/trio/_threads.py

```diff
@@ -397,12 +473,14 @@ def abort(raise_cancel: RaiseCancelT) -> trio.lowlevel.Abort:
             object
         ] = await trio.lowlevel.wait_task_rescheduled(abort)
         if isinstance(msg_from_thread, outcome.Outcome):
+            _decrement_active_threads()
             return msg_from_thread.unwrap()
         elif isinstance(msg_from_thread, Run):
             await msg_from_thread.run()
         elif isinstance(msg_from_thread, RunSync):
             msg_from_thread.run_sync()
         else:  # pragma: no cover, internal debugging guard TODO: use assert_never
+            _decrement_active_threads()
```

Comment:
After looking at this for a bit I agree with @CoolCat467 that the increment/decrement does seem unreliable. if `await msg_from_thread.run()` or `msg_from_thread.run_sync` raises exceptions it won't decrement. So either a contextmanager, or wrapping all the involved lines in `to_thread_run_sync` in a try/finally, would probably be good.